src/core/Microsoft.PowerToys.Settings.UI.Lib/ViewModels/KeyboardManagerViewModel.cs

```diff
@@ -1,241 +1,203 @@
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Windows.Input;
+using Microsoft.PowerToys.Settings.UI.Lib.Helpers;
+using Microsoft.PowerToys.Settings.UI.Lib.Utilities;
+using Microsoft.PowerToys.Settings.UI.Lib.ViewModels.Commands;
+
+namespace Microsoft.PowerToys.Settings.UI.Lib.ViewModels
+{
+    public class KeyboardManagerViewModel : Observable
+    {
+        private const string PowerToyName = "Keyboard Manager";
+        private const string RemapKeyboardActionName = "RemapKeyboard";
+        private const string RemapKeyboardActionValue = "Open Remap Keyboard Window";
+        private const string EditShortcutActionName = "EditShortcut";
+        private const string EditShortcutActionValue = "Open Edit Shortcut Window";
+        private const string JsonFileType = ".json";
+        private const string ProfileFileMutexName = "PowerToys.KeyboardManager.ConfigMutex";
+        private const int ProfileFileMutexWaitTimeoutMilliseconds = 1000;
+
+        private readonly FileSystemWatcher watcher;
+
+        private ICommand remapKeyboardCommand;
+        private ICommand editShortcutCommand;
+        public KeyboardManagerSettings settings;
+        private KeyboardManagerProfile profile;
+        private GeneralSettings generalSettings;
+
+        private Func<string, int> SendConfigMSG { get; }
+
+        private Func<List<KeysDataModel>, int> FilterRemapKeysList { get; }
+
+        public KeyboardManagerViewModel(Func<string, int> ipcMSGCallBackFunc, Func<List<KeysDataModel>, int> filterRemapKeysList)
+        {
+            if (SettingsUtils.SettingsExists(PowerToyName))
+            {
+                // Todo: Be more resilient while reading and saving settings.
+                settings = SettingsUtils.GetSettings<KeyboardManagerSettings>(PowerToyName);
+
+                // Load profile.
+                if (!LoadProfile())
+                {
+                    profile = new KeyboardManagerProfile();
+                }
+            }
+            else
+            {
+                settings = new KeyboardManagerSettings(PowerToyName);
+                SettingsUtils.SaveSettings(settings.ToJsonString(), PowerToyName);
+            }
+
+            if (SettingsUtils.SettingsExists())
+            {
+                generalSettings = SettingsUtils.GetSettings<GeneralSettings>(string.Empty);
+            }
+            else
+            {
+                generalSettings = new GeneralSettings();
+                SettingsUtils.SaveSettings(generalSettings.ToJsonString(), string.Empty);
+            }
+
+            // set the callback functions value to hangle outgoing IPC message.
+            SendConfigMSG = ipcMSGCallBackFunc;
+            FilterRemapKeysList = filterRemapKeysList;
```

Comment:
There is an null reference exception which occurs at the starting when KBM page is opened because `FilterRemapKeysList` is set after `LoadProfile` is called, and `LoadProfile` uses this method. This causes this behavior where list view does not load in the starting:
![image](https://user-images.githubusercontent.com/32061677/89843501-bbdc2700-db2d-11ea-8390-83d37be46529.png)

I think the correct way to fix this is to remove this FilterRemapKeysList function object, and instead just move CombineRemappings and FilterRemapKeysList to this file instead of the xaml.cs file.

Reply:
does this resolve it? https://github.com/microsoft/PowerToys/pull/5821/commits/76daeb822c3f686ebd80171374202fca042b1af6

Reply:
Yes. It would be good to move the Filter method to the ViewModel though if possible. That would be the ideal fix.